bayesian/utility/v2/component_analyses.py

```diff
@@ -168,6 +168,50 @@ def get_batch_ca_data(ecosystem: str, packages: List) -> dict:
     return response
 
 
+def get_batch_ca_vulnerability_data(ecosystem: str, packages: List) -> dict:
+    """Fetch package details for component analyses."""
+    logger.debug('Executing get_batch_ca_data')
+    started_at = time.time()
+
+    response = None
+    semver_packages = []
+    pseudo_version_packages = []
+
+    # Need to seperate semver and pseudo verion packages for golang
+    if (ecosystem == "golang"):
+        for p in packages:
+            if p['is_pseudo_version']:
+                pseudo_version_packages.append(p)
+            else:
+                semver_packages.append(p)
+    else:
+        semver_packages = packages
+
+    graph_data_fetcher = []
+    if len(semver_packages) > 0:
+        get_semver_data = functools.partial(GraphAnalyses.get_batch_ca_vulnerability_data,
+                                            ecosystem)
+        graph_data_fetcher = list(_fetcher_in_batches(get_semver_data, semver_packages))
+
+    if len(pseudo_version_packages) > 0:
+        get_pseudo_data = functools.partial(GraphAnalyses.get_batch_ca_data_for_pseudo_version,
+                                            ecosystem)
+        graph_data_fetcher += list(_fetcher_in_batches(get_pseudo_data, pseudo_version_packages))
+
+    response = {
+        "result": {
+            "data": []
+        }
+    }
+    result = EXECUTOR.map(lambda f: f(), graph_data_fetcher)
```

Comment:
Do we know whats the latency we add by adding all this lambda, functools bla bla compared to simply running an array and calling gremlin in batches?

Reply:
Nope.I modeled the api according the last CA one.I asumed best practice was used.so incroporated them.

Reply:
here we are working on a tight response time.any added latency should be removed. so please check this behaviour with and without all this decoration

Reply:
after some research it looks like the calls here are happening concurrenlty so it will be faster than calling gremlin in batches.